### DISCUSSION TITLE
blog/in-memory-caching-in-aspnet-core/

### DISCUSSION THREAD

**Discussion by giscus** [codewithmukesh/discussions#14] [General]
# blog/in-memory-caching-in-aspnet-core/

Caching! In this article, we will explore the fundamentals of caching and demonstrate how to implement in-memory caching in ASP.NET Core applications. We'll start by building a simple endpoint to set and retrieve cache entries from the in-memory cache. Following that, we'll create another endpoint to showcase the noticeable performance improvement—over 80% faster—achieved through caching. We will also cover some important topics like cache invalidation.

https://codewithmukesh.com/blog/in-memory-caching-in-aspnet-core/?utm_source=linkedin

**Comment by imraja** [comment]
Hi Mukesh, Thanks for the article. I see "SetAbsoluteExpiration" mentioned two times in the same code block, and I think you have to replace it with "SetSlidingExpiration". Please check.

**Reply by iammukeshm** [reply]
Thanks for highlighting this! Taken care of it now.

**Comment by VahidN** [comment]
It's better to use the AOP/Decorator pattern techniques to remove the repetitive code of caching from our controllers or [EFCoreSecondLevelCacheInterceptor](https://github.com/VahidN/EFCoreSecondLevelCacheInterceptor) library can help you to eliminate the duplicate code of caching in your services and API's.

**Reply by iammukeshm** [reply]
Yes, in one of the next article I will be showing how to integrating Caching as a behavior in CQRS/MediatR applications. But, yeah decorator pattern will make it even for flexible.

**Comment by hectorlaris** [comment]
Mukesh thank you for sharing this content, it has been very useful.

**Comment by AkshatTiwari93** [comment]
Hi Mukesh, Thanks for the article. However when I looked into the use of SetSize() method I found that it will only work if SizeLimit option is specified in the AddMemoryCache() declaration otherwise it is ignored. So I find it unnecessary to specify it in GetAll() method.
Great Work Otherwise.

**Comment by Morteza-Motayari** [comment]
thank you very much bro 
your explanation are very clear and understandable for my opinion its better than Microsoft Explanation
how much I appreciate you I can't valuate it

**Comment by Lakshmi-R** [comment]
thanks for sharing the wounderful resources.

**Comment by TrongNghia-17** [comment]
This article is very helpful, thank you very much Mukesh.
